contracts/vaults/VaultPauser/RibbonVaultPauser.sol

```diff
@@ -0,0 +1,249 @@
+
+    function getPausePosition(address _vaultAddress, address _userAddress)
+        external
+        view
+        returns (PauseReceipt memory)
+    {
+        return pausedPositions[_vaultAddress][_userAddress];
+    }
+
+    /************************************************
+     *  SETTERS
+     ***********************************************/
+
+    /**
+     * @notice Sets the new keeper
+     * @param _newKeeper is the address of the new keeper
+     */
+    function setNewKeeper(address _newKeeper) external onlyOwner {
+        require(_newKeeper != address(0), "!newKeeper");
+        keeper = _newKeeper;
+    }
+
+    /**
+     * @notice add vault into registered vaults
+     * @param _vaultAddress is the address of the new vault to be registered 
+     */
+    function addVault(address _vaultAddress) external onlyOwner {
+        registeredVaults[_vaultAddress] = true;
+    }
+
+    /************************************************
+     *  VAULT OPERATIONS
+     ***********************************************/
+
+    /**
+     * @notice pause position from vault by redeem all the shares from vault to Pauser
+     * @param _account user's address
+     * @param _amount the amount of shares
+     */
+    function pausePosition(address _account, uint256 _amount)
+        external
+        override
+    {
+        address currentVaultAddress = msg.sender;
+        IRibbonThetaVault currentVault = IRibbonThetaVault(currentVaultAddress);
+
+        // check if vault is registered
+        require(
+            registeredVaults[currentVaultAddress],
+            "Vault is not registered"
+        );
+
+        PauseReceipt storage pausedPosition =
+            pausedPositions[currentVaultAddress][_account];
+
+        // check if position is paused
+        require(
+            pausedPosition.shares == 0 && pausedPosition.round == 0,
+            "Position is paused"
+        );
+
+        uint16 round = currentVault.vaultState().round;
+
+        require(_amount < type(uint128).max, "_amount overflow");
+
+        pausedPositions[currentVaultAddress][_account] = PauseReceipt({
+            round: round,
+            shares: uint128(_amount)
+        });
+
+        emit Pause(_account, currentVaultAddress, _amount, round);
+
+        // transfer from user to pauser
+        IERC20(currentVaultAddress).safeTransferFrom(
+            _account,
+            address(this),
+            _amount
+        );
+
+        currentVault.initiateWithdraw(_amount);
+    }
+
+    /**
+     * @notice resume user's position into vault by making a deposit
+     * @param _vaultAddress vault's address
+     */
+    function resumePosition(address _vaultAddress) external override {
+        IRibbonThetaVault currentVault = IRibbonThetaVault(_vaultAddress);
+
+        // check if vault is registered
+        require(registeredVaults[_vaultAddress], "Vault is not registered");
+
+        address currentUser = address(msg.sender);
```

Comment:
Can remove the `address(`, also can probably just use `msg.sender` instead of `currentUser`